torch/distributed/algorithms/ddp_comm_hooks/optimizer_overlap_hooks.py

```diff
@@ -12,25 +12,32 @@ class _OptimizerHookState(object):
     Currently contains only optimizer class which must have a method `step_param`.
     """
 
-    __slots__ = ["functional_optimizer"]
+    __slots__ = ["functional_optimizer", "params_to_optimize"]
 
     def __init__(
-        self, functional_optim_cls, *functional_optim_args, **functional_optim_kwargs
+        self, functional_optim_cls, *functional_optim_args, params=None, **functional_optim_kwargs
     ):
         self.functional_optimizer = create_functional_optim(
             functional_optim_cls,
             *functional_optim_args,
             **functional_optim_kwargs,
         )
         self._check_valid_functional_optim()
+        self._set_params_to_optimize(params)
 
     @classmethod
-    def from_functional_optim(cls, functional_optim):
+    def from_functional_optim(cls, functional_optim, params=None):
         opt_hook_state_inst = cls.__new__(cls)  # Does not call __init__
         opt_hook_state_inst.functional_optimizer = functional_optim
         opt_hook_state_inst._check_valid_functional_optim()
+        opt_hook_state_inst._set_params_to_optimize(params)
         return opt_hook_state_inst
 
+    def _set_params_to_optimize(self, params):
+        self.params_to_optimize = params
```

Comment:
Nit: This can be simplified to:

```
if params is not None:
    self.params_to_optimize = set(params)
```

Reply:
This leads to `self.params_to_optimize` not being defined in the case it is None, but we access it on L60, but we can change that to check `hasattr` as well.